processing/src/main/java/org/apache/druid/segment/FilteredStorageAdapter.java

```diff
@@ -45,26 +42,21 @@ public FilteredStorageAdapter(final StorageAdapter adapter, final DimFilter filt
   }
 
   @Override
-  public Sequence<Cursor> makeCursors(
-      @Nullable Filter filter,
-      Interval interval,
-      VirtualColumns virtualColumns,
-      Granularity gran,
-      boolean descending,
-      @Nullable QueryMetrics<?> queryMetrics
-  )
+  public CursorHolder makeCursorHolder(CursorBuildSpec spec)
   {
-    final Filter andFilter;
-    if (filter == null) {
+    final CursorBuildSpec.CursorBuildSpecBuilder buildSpecBuilder = CursorBuildSpec.builder(spec);
+    final Filter newFilter;
+    if (spec.getFilter() == null) {
       if (filterOnDataSource != null) {
```

Comment:
`FilteredDataSource` allows null for some reason, so i just marked it nullable and added a check for now. I don't really understand why we would allow filter to be null though, that doesn't seem very useful, so it might be worth looking into changing it to ensure not null in a follow-up